Projections added to python client (#451)
hazelcast/hazelcast-python-client: docs/using_python_client_with_hazelcast_imdg.rst

    # Average age is 30
    print("Average age is %f" % average_age)

Projections
~~~~~~~~~~~~~~~~~

> Projections feature provides three projection functions; ``single_attribute``, 
> ``multi_attribute`` and ``identity``, on top of Hazelcast ``Map`` entries. Their
> performance is high since they run in parallel for each partition and are 
> highly optimized for speed and low memory consumption.
> 
> These projections are used with the ``map.project`` function, which takes an
> optional predicate argument.
> 
> See the following example.

Reviewer: ```suggestion
There are cases where instead of sending all the data returned by a query
from the server, you want to transform (strip down) each result object in order
to avoid redundant network traffic.

For example, you select all employees based on some criteria, but you just
want to return their name instead of the whole object. It is easily doable
with the Projections.

The ``projection`` module provides three projection functions:

- ``single_attribute``: Extracts a single attribute from an object and returns
  it.
- ``multi_attribute``: Extracts multiple attributes from an object and returns
  them as a ``list``.
- ``identity``: Returns the object as it is.

These projections are used with the ``map.project`` function, which takes an
optional predicate argument.

See the following example.
```
Let's replace the docs with this.

.. code:: python

    import hazelcast

    from hazelcast.core import HazelcastJsonValue

>     from hazelcast.projection import single_attribute, multi_attribute
>     from hazelcast.predicate import greater_or_equal

Reviewer: ```suggestion
    from hazelcast.predicate import greater_or_equal
    from hazelcast.projection import single_attribute, multi_attribute
```
Change the import order to match alphabetical order

    client = hazelcast.HazelcastClient()
    employees = client.get_map("employees").blocking()

>     employees.put(1, HazelcastJsonValue('{"Age": 23, "Height": 180, "Weight": 60}'))

Reviewer: ```suggestion
    employees.put(1, HazelcastJsonValue('{"Age": 23, "Height": 180, "Weight": 60}'))
    employees.put(2, HazelcastJsonValue('{"Age": 21, "Height": 170, "Weight": 70}'))
```
Let's put one more employee to show that predicates used below works

>     employee_age = employees.project(single_attribute("Age"))

Reviewer: ```suggestion
    employee_ages = employees.project(single_attribute("Age"))
```
As this will return a list of ages

    # Prints:

>     # The employee age is 23
>     print("The employee age is %d" % employee_age)

Reviewer: Have you run this code? I believe the variable is a list, not a single objects. Please correct the output if it is something else.

```suggestion
    # The ages of employees are [23, 25]
    print("The ages of employees are %s" % employee_age)
```

Reviewer: Same for others, please run the and verify they produce the output you wrote

optional predicate argument.

See the following example.

.. code:: python

    import hazelcast

    from hazelcast.core import HazelcastJsonValue
    from hazelcast.predicate import greater_or_equal
    from hazelcast.projection import single_attribute, multi_attribute

    client = hazelcast.HazelcastClient()
    employees = client.get_map("employees").blocking()

    employees.put(1, HazelcastJsonValue('{"Age": 23, "Height": 180, "Weight": 60}'))
    employees.put(2, HazelcastJsonValue('{"Age": 21, "Height": 170, "Weight": 70}'))

    employee_ages = employees.project(single_attribute("Age"))
    # Prints:
    # The ages of employees are [21, 23]
    print("The ages of employees are, ", employee_ages)

    # Run Single Attribute With Predicate
    employee_ages = employees.project(single_attribute("Age"), greater_or_equal("Age", 23))
    # Prints:
    # The employee age is 23
    print("The employee age is: " % employee_ages[0])

    # Run Multi Attribute Projection
    employee_multi_attribute = employees.project(multi_attribute("Age", "Height"))
    # Prints:
    # Employee 1 age and height: [23, 180] Employee 2 age and height: [21, 170]

>     print("Employee 1 age and height: ", employee_multi_attribute[1], " Employee 2 age and height: " employee_multi_attribute[0])

Reviewer: ```suggestion
    print("Employee 1 age and height: %s Employee 2 age and height: %s" % (employee_multi_attribute[0], employee_multi_attribute[1]))
```
No need to change the printed order just above as it is not deterministic. Let's just use 0,1 instead of 1,0 to make it more intuitive
